app/assets/javascripts/beak/widgets/widget-controller.coffee

```diff
@@ -271,24 +278,15 @@ defaultWidgetMixinFor = (widgetType, x, y, countByType) ->
     when "plot"     then { bottom: y + 160, right: x + 200, autoPlotOn: true, display: "Plot #{countByType(widgetType) + 1}", legendOn: false, pens: [], setupCode: "", updateCode: "", xAxis: "", xmax: 10, xmin: 0, yAxis: "", ymax: 10, ymin: 0, exists: false }
     when "textBox"  then { bottom: y +  60, right: x + 180, color: 0, display: "", fontSize: 12, transparent: true }
     else throw new Error("Huh?  What kind of widget is a #{widgetType}?")
+# coffeelint: enable=max_line_length
 
 # [T <: Widget] @ Array[String] => T => T => Boolean
 widgetEqualsBy = (props) -> (w1) -> (w2) ->
   { eq } = tortoise_require('brazier/equals')
-  locationProps = ['bottom', 'left', 'right', 'top']
-  propEq        = (prop) -> eq(w1[prop])(w2[prop])
-  w1.type is w2.type and locationProps.concat(props).every(propEq)
-
-buttonProps   = ['buttonKind', 'disableUntilTicksStart', 'forever', 'source']
-chooserProps  = ['choices', 'display', 'variable']
-inputBoxProps = ['boxedValue', 'variable']
-monitorProps  = ['display', 'fontSize', 'precision', 'source']
-outputProps   = ['fontSize']
-plotProps     = [ 'autoPlotOn', 'display', 'legendOn', 'pens', 'setupCode', 'updateCode', 'xAxis', 'xmax', 'xmin', 'yAxis', 'ymax', 'ymin']
-sliderProps   = ['default', 'direction', 'display', 'max', 'min', 'step', 'units', 'variable']
-switchProps   = ['display', 'on', 'variable']
-textBoxProps  = ['color', 'display', 'fontSize', 'transparent']
-viewProps     = ['dimensions', 'fontSize', 'frameRate', 'showTickCounter', 'tickCounterLabel', 'updateMode']
-# coffeelint: enable=max_line_length
+  propEq = (prop) ->
+    v1 = w1[prop]
+    v2 = w2[prop]
+    ((v1 is null or v1 is undefined) and (v2 is null or v2 is undefined)) or eq(v1)(v2)
```

Comment:
How does it happen that we have a widget with one of its properties un-defined?

Reply:
Oof, I should've written this down.  I want to say it was something like when changing the type of an input widget during an edit?  I don't remember the scenario 100%, but I do remember it seemed like a normal legit bug to me when I found it.